rgw/v2/tests/s3_swift/reusable.py

```diff
@@ -293,6 +293,43 @@ def enable_versioning(bucket, rgw_conn, user_info, write_bucket_io_info):
     else:
         raise TestExecError("version enable failed")
 
+def generate_totp(seed):
+    cmd='oathtool -d6 --totp %s' %seed
+    totp_token=utils.exec_shell_cmd(cmd)
+    return totp_token.rstrip("\n")
+
+def enable_mfa_versioning(bucket, rgw_conn, SEED, serial, user_info, write_bucket_io_info):
+    log.info('bucket MFA and versioning test on bucket: %s' % bucket.name)
+    bucket_versioning = s3lib.resource_op({'obj': rgw_conn,
+                                           'resource': 'BucketVersioning',
+                                           'args': [bucket.name]})
+    # checking the versioning status
+    version_status = s3lib.resource_op({'obj': bucket_versioning,
+                                        'resource': 'status',
+                                        'args': None
+                                        })
+    if version_status is None:
```

Comment:
it looks like you need to error out if this condition fails. can you raise an exception. probably a new exception !!

Reply:
sure, raised a new exception MFAVersionError